packages/tuono/src/ssr/utils.ts

```diff
@@ -0,0 +1,36 @@
+function concatArrayBuffers(chunks: Array<Uint8Array>): Uint8Array {
+  const result = new Uint8Array(chunks.reduce((a, c) => a + c.length, 0))
+  let offset = 0
+  for (const chunk of chunks) {
+    result.set(chunk, offset)
+    offset += chunk.length
+  }
+  return result
+}
+
+async function streamToArrayBuffer(
+  stream: ReadableStream<Uint8Array>,
+): Promise<Uint8Array> {
+  const chunks: Array<Uint8Array> = []
+
+  // @ts-expect-error ReadableStream has AsyncIterator https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream#async_iteration
+  // eslint-disable-next-line @typescript-eslint/await-thenable
+  for await (const chunk of stream) {
+    chunks.push(chunk as Uint8Array)
+  }
+
+  return concatArrayBuffers(chunks)
+}
+
+/**
+ * This function awaits for the whole stream before returning the string.
+ *
+ * NOTE: we should improve the bond between the custom V8 runtime and the
+ * renderToReadableStream React function to return a stream directly to the client.
+ */
+export async function streamToString(
+  stream: ReadableStream<Uint8Array>,
+): Promise<string> {
+  const buffer = await streamToArrayBuffer(stream)
+  return new TextDecoder().decode(buffer)
+}
```

Comment:
I would consider removing `concatArrayBuffers` and `streamToArrayBuffer` and merge their code all in `streamToString` function.

Right now the functions are pretty simple so there is no need to split the code in 3 parts. WDYT?

Reply:
Yeah. I left separate since it is more verbose this way. This code does very weird things, very uncommon. I think a bit more meaning could be helpful while reading this code in the future

Reply:
the final function has only two  rows so I would remove at least one, but I get your point we can leave this as it is!